src/mesh/mesh.cxx

```diff
@@ -10,8 +9,6 @@
 #include <bout/boutcomm.hxx>
 #include <bout/output.hxx>
 
-#include "impls/bout/boutmesh.hxx"
```

Comment:
I think this is probably what's breaking the static builds as the `RegisterMesh<BoutMesh> registermeshbout{"bout"}` variable won't be seen anywhere.